diff --git a/TiffWriter.cpp b/TiffWriter.cpp
index 6c837e7..472def7 100644
--- a/TiffWriter.cpp
+++ b/TiffWriter.cpp
@@ -240,8 +240,10 @@ bool TiffWriter::writeBitonalOrIndexed8Image(const TiffHandle& tif, const QImage
   }
 
   if (image.format() == QImage::Format_Indexed8) {
-    TIFFSetField(tif.handle(), TIFFTAG_COMPRESSION,
-                 uint16(QSettings().value("settings/color_compression", COMPRESSION_LZW).toInt()));
+    uint16 compress = (photometric == PHOTOMETRIC_PALETTE) ?
+                      COMPRESSION_LZW :
+                      uint16(QSettings().value("settings/color_compression", COMPRESSION_LZW).toInt());
+    TIFFSetField(tif.handle(), TIFFTAG_COMPRESSION, compress);
   } else {
     TIFFSetField(tif.handle(), TIFFTAG_COMPRESSION,
                  uint16(QSettings().value("settings/bw_compression", COMPRESSION_CCITTFAX4).toInt()));
